drivers/hmis/lib/tasks/graphql.rake

```diff
@@ -1,7 +1,21 @@
-task dump_graphql_schema: :environment do
-  # Replace 'MySchema' with your schema's Ruby class name.
+require 'digest/sha1'
+
+desc 'Update schema.graphql dump'
+task dump_graphql_schema: [:environment, 'log:info_to_stdout'] do
   schema_definition = HmisSchema.to_definition
   schema_path = Rails.root.join('drivers', 'hmis', 'app', 'graphql', 'schema.graphql')
+
+  old_sha = Digest::SHA1.hexdigest(File.read(schema_path))
+
   File.write(schema_path, schema_definition)
-  puts "Updated #{schema_path}"
+
+  new_sha = Digest::SHA1.hexdigest(File.read(schema_path))
+  if old_sha != new_sha
+    puts "Updated #{schema_path}"
+    # Needed for GitHub Actions check:
+    # Exit with status code 1 to indicate that the schema has changed
+    exec('false')
```

Comment:
can you not just call `exit(false)` or `raise 'GraphQL Schema out of sync!` here?

Reply:
`exit(false)` exits  with status code zero. Raising an error would fine, but the better way to exit gracefully with an error is using `abort`. I'll switch it to that https://github.com/greenriver/hmis-warehouse/pull/2363/commits/66902215cb2c285ab4ef263c1d42e16412baff8a

Reply:
Thanks!

Reply:
I was avoiding when I first wrote this, because errors print a huge stack trace for me because I don't have the Slack exception notifier set up (so it uses "www.unknown.com"). But that's a me-problem 🙂 
Do you know why we use [unknown.com here](https://github.com/greenriver/hmis-warehouse/blob/production/config/exception_notifier.yml#L9-L140)? I assume it was put there for deployment reasons, so I made a change to ignore it in development: https://github.com/greenriver/hmis-warehouse/pull/2363/commits/d6a0c30ee385a10c815cb731fec7fb83f8d5f5fb

Reply:
That's one I hadn't noticed.  I believe we used to have strings "unknown" but maybe we needed a "valid" url.

Reply:
I think something is wrong if it can't find the RAILS_ENV though. It should be `test` in this scenario.
ENV.fetch('RAILS_ENV') { 'unknown' }

Reply:
I think it's finding the RAILS_ENV just fine, "unknown" is just the fallback value? unless I'm misreading?

Reply:
That's exactly what's happening.  Just not sure why you'd be seeing the stack trace if you have a RAILS_ENV.  The fallback is probably for a deployment container where the ENV may not be set.

Reply:
I'm seeing it because I don't have `EXCEPTION_WEBHOOK_URL` set for my local dev environment, which leads to this [else statement](https://github.com/greenriver/hmis-warehouse/blob/6e3d3f32f3183b3cb1056acbe80058e4d14d3bea/config/exception_notifier.yml#L9) which fills in the config with the "unknown.com" url. It doesn't have anything to do with the RAILS_ENV, which is development. Then the slack notifier gets enabled over here https://github.com/greenriver/hmis-warehouse/commit/d6a0c30ee385a10c815cb731fec7fb83f8d5f5fb

Reply:
I guess really it should just be "else if environment is production" [here](https://github.com/greenriver/hmis-warehouse/blob/6e3d3f32f3183b3cb1056acbe80058e4d14d3bea/config/exception_notifier.yml#L9), then. I can make that change if you agree with that!

Reply:
Thanks, I think that's better.